### PR TITLE
create a view from auth and content server datasets

### PR DESCRIPTION
**Pull request by irrationalagent**
fixes https://github.com/mozilla/bigquery-etl/issues/135

This creates a similar dataset to https://github.com/mozilla/bigquery-etl/blob/master/sql/fxa_all_events_v1.sql (minus the bounce events) but with columns extracted from `jsonPayload.fields.user_properties` and `jsonPayload.fields.event_properties`

Jeff, I noticed that querying for `fxa_auth_events_v1` requires a filter on timestamp, so I added an arbitrary one here. Do you know if there is  a way I could remove that, so the dataset can go back as far as possible?

Also, wasn't sure if any tests were needed here.

### CODE REVIEW

**Comment by relud** [comment]
> `fxa_auth_events_v1` requires a filter on timestamp

When you include `timestamp` it should be possible to set the filter when querying the view, instead of in the view. The requirement is meant as a cost control measure, it's fine to lift it for these tables if we're sure they don't need it.

File: sql/fxa_content_auth_events_v1.sql [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    jsonPayload.fields.os_version,
+    jsonPayload.fields.country,
+    jsonPayload.fields.language,
+    jsonPayload.fields.user_id,
+    jsonPayload.fields.user_properties,
+    jsonPayload.fields.event_properties,
+    timestamp,
+    receiveTimestamp
+  FROM
+    `moz-fx-data-derived-datasets.telemetry.fxa_auth_events_v1`
```
Comment (jklukas) [inline sql/fxa_content_auth_events_v1.sql]: Added in explicit project prefix to the table names, which is necessary in view definitions.

File: sql/fxa_content_auth_events_v1.sql [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  --
+  unioned AS (
+    SELECT * FROM auth
+    UNION ALL
+    SELECT * FROM content
+  )
+  --
+SELECT
+  * EXCEPT (user_properties, event_properties),
+  REPLACE(JSON_EXTRACT(user_properties, '$.utm_term'),"\"","") AS utm_term,
```
Comment (jklukas) [inline sql/fxa_content_auth_events_v1.sql]: Delayed the json extraction to after the union to avoid duplicating the logic.

**Comment by jklukas** [comment]
@irrationalagent - I pushed a change to refactor a bit and to fix some tiny details. I've created the view, so feel free to test it out and let me know if this is good to merge or if you'd like to make additional changes.

**Comment by irrationalagent** [comment]
Thanks for the cleanup! - ill write some queries and post back later today.

**Comment by irrationalagent** [comment]
Comparing [a query from this view](https://datastudio.google.com/u/0/reporting/1wip1u10YmEUYC6SKFaJRoJ1br7bNtMlj/page/Hiwp) matches a similar one from [actual amplitude](https://analytics.amplitude.com/mozilla-corp/chart/pa0y0l6), so I think we're good here in terms of accuracy.

I also think this contains all the measurements we need to make the trailhead dashboard per the current spec. There's an open question of whether we will need more historical data to compute e.g. baselines for retention, but let's punt on that for now.

So I think we're good to merge.

**Pull request merged**